app/models/hotel.rb

```diff
@@ -1,3 +1,7 @@
 class Hotel < ApplicationRecord
   has_many :rooms
+
+  def guest_list
+    Guest.distinct.joins(rooms: :hotel).where(hotel: {name: self.name}).pluck(:name).join(", ")
```

Comment:
This is great and correct! Another option would be to start at Hotel and move the other direction:
`Hotel.joins(rooms: :guests).where("rooms.hotel_id = 2").distinct("guests.name").pluck("guests.name")`

However, an even simpler option is to use associations. It's possible in AR to use the `through` associations to chain associations and leverage relationships set up for a resource a model is already associated to. In this case, hotel is associated to rooms, and rooms has associations set up to guests. So, you can add an association to hotel like this: `has_many :guests, through: :rooms`
Then, in your guest_list method, you can simply use `guests.distinct` to grab the guests from a given hotel! Very simple. Very nifty.